### ISSUE TITLE
First steps

### ISSUE THREAD

**Issue by wilzbach** [dlang-community/discussions#2]
From the [pain](https://github.com/dlang-tour/dyaml-dlang-tour/pull/6) of the Dlang Tour maintainers the following discussion started:

>> Maybe it's not a bad idea to have a couple of important D packages like this one a community namespace, s.t. it doesn't depend on one maintainer?

> I've also thought about that. Maybe a common github organization where all prominent members of the community have merge rights, though individual projects would still be driven by their authors. Similar to phobos experimental, but not tied to dmd releases, and with more lax requirements for entry and respectively a bit less guarantees.

I [know](https://github.com/dlang/dmd/pull/6312#issuecomment-266274778) that @MartinNowak has experienced these problems too:

> I became of a maintainer for all those packages, though I certainly will only keep a curated list.

So I went ahead and created such an organization (see #1 for the name) and invited @ZombineDev and @MartinNowak as owners.

So the first steps are:

- [x] move dyaml to this org (@ZombineDev - I'll assign this to you as you have registered the clone on DUB)
- [x] move drepl/drepl to this org (one of the "lost" projects I can think of atm)
- invite more trusted D community members ;-)

**Comment by ZombineDev** [comment]
Thanks for going forward with this @wilzbach! I'll try to find some time to move the dyaml repo soon.

**Comment by MartinNowak** [comment]
To me it looks a bit like your trying to solve points that are better addressed by code.dlang.org 2.0, namely prioritizing packages and having a few admin options, e.g. to rename a package and requesting to become the owner of a package.

**Comment by MartinNowak** [comment]
It's already possible to add people to existing repos btw., and distributed is usually more resilient than centralized.
Just the first thoughts on that.

**Comment by kozhevnikov-peter** [comment]
I and my company agree with the offer and are ready to take an active part in this project. I really like Dlang and I try to popularize it. But when the code ceases to be supported, many leave the Dlang community.

**Comment by wilzbach** [comment]
> I and my company agree with the offer and are ready to take an active part in this project.

Awesome :)
I have sent out an invitation!

**Comment by kozhevnikov-peter** [comment]
I will create a repository yamld after some fixes (I plan to transfer the test code from the working code).

**Comment by kozhevnikov-peter** [comment]
We do not mind if I rename dyaml in yaml-d? Or do you have any suggestions?

**Comment by wilzbach** [comment]
> We do not mind if I rename dyaml in yaml-d?

Not really - I am quite aware of the fact the new DUB name needs to be different.

> Or do you have any suggestions?

Why not just `yaml`? AFAICT it's still available.

**Comment by kozhevnikov-peter** [comment]
@wilzbach  Thanks, I'll make yaml.

**Comment by leandro-lucarella-sociomantic** [comment]
We [forked D-YAML](https://travis-ci.org/sociomantic-tsunami/D-YAML) too because we thought it was dead. If this project is resurrected we will probably be willing to contribute if contributions are timely reviewed and merged.

**Comment by wilzbach** [comment]
> We forked D-YAML too because we thought it was dead. If this project is resurrected we will probably be willing to contribute if contributions are timely reviewed and merged.

Wow. @leandro-lucarella-sociomantic I invited you to the dlang-community as well. 

>  if contributions are timely reviewed and merged.

Please feel free to add interested coworkers of yours as well.


@ZombineDev @kozhevnikov-peter as nothing happened, I took the liberty of moving `dyaml-dlang-tour` to `dlang-community`: https://github.com/dlang-community/yaml
License to do amazing things with it is granted ;-)

**Comment by leandro-lucarella-sociomantic** [comment]
I created a fork of hardbored-mod, we also have a fork for it, but maybe I could just move ours, as it already got some contributions. This way people acceding the old URL will be automatically redirected. What do you think?

**Comment by BBasile** [comment]
I'm in favor. I already consider your fork (if i understand well you talk about this one https://github.com/sociomantic-tsunami/harbored-mod) as my official git origin now.

**Comment by leandro-lucarella-sociomantic** [comment]
Yup. Then we'll discuss a bit internally but we'll probably move the repo here. I'll remove the freshly done fork to avoid people sending PRs there.

**Comment by wilzbach** [comment]
> I'm in favor. I already consider your fork (if i understand well you talk about this one https://github.com/sociomantic-tsunami/harbored-mod) as my official git origin now.

+1 - we just need to improve DUB, s.t. people looking for documentation generators don't find the outdated harbored-mod first. Maybe a simple rename can do the trick?

**Comment by leandro-lucarella-sociomantic** [comment]
> To me it looks a bit like your trying to solve points that are better addressed by code.dlang.org 2.0, namely prioritizing packages and having a few admin options, e.g. to rename a package and requesting to become the owner of a package.

I think this will be best (taking over ownership of dub projects instead of renaming them each time they are abandoned)

> It's already possible to add people to existing repos btw., and distributed is usually more resilient than centralized.

But about this, is true that it makes the situation more confusing if the original author stops developing and someone else wants to take over. Having a moving project is more annoying. If there is an org for this where people can come and go but projects are still alive, I think there is value in it, even when I agree that distributed is better. Maybe for the state D it is now, where is very common for this to happen (people abandoning projects and other people maybe wanting to take over) is good to have this kind of org.

**Comment by BBasile** [comment]
Has someone ever tried to contact @kiith-sa at least ?

**Comment by leandro-lucarella-sociomantic** [comment]
I'm pretty sure we tried, right @nemanja-boric-sociomantic?

**Comment by BBasile** [comment]
I think i know what happened. He created a documentation site with all the D projects from DUB rendered by hmod. Then as nobody cared, it was hard to swallow and he left _"the scene"_. Also i remember a message on the NG where he explained that he wanted to focus on his studies.

I anyone wants to try again, his mail is kiithsacmp :hat: gmail.com.

**Comment by nemanja-boric-sociomantic** [comment]
> I'm pretty sure we tried, right @nemanja-boric-sociomantic?

Yeah, I think I did contact him last year, I will also try again now.

**Comment by nemanja-boric-sociomantic** [comment]
Sent email.

**Comment by leandro-lucarella-sociomantic** [comment]
[harbored-mod](https://github.com/dlang-community/harbored-mod) transferred!

**Comment by BBasile** [comment]
In case we couldn't get the ownership of dyaml and hmod i still think that it would be better to create new repositories and init them with the content of the forks. Forks are finally only a specific GH thing and if the original are dead then there no need to maintain the link.

**Comment by kiith-sa** [comment]
TLDR: got a job, not much free time. Will look at transferring ownership this weekend (who to? - never did that with GitHub). Would like to eventually fix issues w/ D-YAML/other projects, but unlikely in near future/current job, so it's better if someone else steps in.

**Comment by wilzbach** [comment]
> Will look at transferring ownership this weekend (who to? - never did that with GitHub). 

Awesome! I invited you to this GH organization (D Community hub). After you accepted the invitation, go to the settings page of your repo, e.g. https://github.com/kiith-sa/D-YAML

![image](https://cloud.githubusercontent.com/assets/4370550/25892825/ee672154-3575-11e7-8b3f-36b06d0cdc51.png)

And then finally just enter "dyaml" and "dlang-community":

![image](https://cloud.githubusercontent.com/assets/4370550/25892863/07a8ffa2-3576-11e7-9776-282ab7adc5e5.png)

**Comment by wilzbach** [comment]
> harbored-mod transferred!
 
@leandro-lucarella-sociomantic great, but should we delete [the fork](https://github.com/dlang-community/harbored-mod), s.t. @kiith-sa can transfer the original repository?

**Comment by leandro-lucarella-sociomantic** [comment]
OK, I backed up the issues just in case. Go ahead! We can then push our changes on top of it (and migrate issues if needed) @nemanja-boric-sociomantic FYI

**Comment by nemanja-boric-sociomantic** [comment]
Ack. I would migrate the issues, since they are useful :-)

**Comment by nemanja-boric-sociomantic** [comment]
Looks like the fork is still in the place (should be deleted). I don't have permissions to do it. cc @leandro-lucarella-sociomantic @wilzbach

**Comment by wilzbach** [comment]
> Looks like the fork is still in the place (should be deleted). I don't have permissions to do it. cc @leandro-lucarella-sociomantic @wilzbach

You should have them now ;-)

**Comment by nemanja-boric-sociomantic** [comment]
Thanks! Deleted.

**Comment by nemanja-boric-sociomantic** [comment]
I'll send another email to @kiith-sa, just in case he has the github notifications turned off.

**Comment by kiith-sa** [comment]
harbored-mod transferred. Was unable to transfer the D-YAML repo; got error: `dlang-community already has a repository in the kiith-sa/D-YAML network `, possibly due to the `yaml` repo?

**Comment by wilzbach** [comment]
> harbored-mod transferred. Was unable to transfer the D-YAML repo; got error: dlang-community already has a repository in the kiith-sa/D-YAML network, possibly due to the yaml repo?

Ah damn. I moved `yaml` to https://github.com/wilzbach/yaml temporarily, would you be so kind to try again?

**Comment by kiith-sa** [comment]
Done.

**Comment by BBasile** [comment]
New TODO for dyaml and hmod: PR with commits from the up-to-date forks.

**Comment by BBasile** [comment]
Which D-YAML should be used to update the original ? I see that they have all diverged.

**Comment by BBasile** [comment]
Also, @kozhevnikov-peter, shouldn't you and your co-worker handle D-YAML ?

**Comment by kozhevnikov-peter** [comment]
@BBasile, Sorry! I'm a little confused already. Which repository do you need to work with? Delete yaml-d? I discovered the problem. Dumper.dump () does not work properly (problem with cast (ubyte[])). I have already corrected yaml-d.

**Comment by BBasile** [comment]
One of this [fork](https://github.com/dlang-community/D-YAML/network) must be used to create a PR (from ~master to ~master) for the original D-YAML which we own now. The Q is which one, beause they've all diverged, though they seem to contain more or less the same fixes.

**Comment by wilzbach** [comment]
I am closing this issue as [D-YAML](https://github.com/dlang-community/D-YAML) has been moved to dlang-community and I think we have managed to do our first steps :)

A huge thanks to all who made this organization and transition possible.
In case you ever have a question, don't hesitate to open an issue at this repo.
